adapters.go

```diff
@@ -316,30 +392,47 @@ func (a *whisperAdapter) SendPublic(ctx context.Context, chatName, chatID string
 		return nil, errors.Wrap(err, "failed to encode message")
 	}
 
-	newMessage := whisper.NewMessage{
-		TTL:       whisperTTL,
-		Payload:   encodedMessage,
-		PowTarget: whisperPoW,
-		PowTime:   whisperPoWTime,
+	wrappedMessage, err := a.wrapMessageV1(encodedMessage)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to wrap message")
+	}
+
+	messageSpec, err := a.protocol.BuildPublicMessage(a.privateKey, wrappedMessage)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to build public message")
+	}
+
+	newMessage, err := a.messageSpecToWhisper(messageSpec)
+	if err != nil {
+		return nil, err
 	}
 
 	_, err = a.transport.SendPublic(ctx, newMessage, chatName)
 	if err != nil {
 		return nil, err
 	}
 
-	return protocol.MessageID(&a.privateKey.PublicKey, encodedMessage), nil
+	return protocol.MessageID(&a.privateKey.PublicKey, wrappedMessage), nil
 }
 
 // SendPublicRaw takes encoded data, encrypts it and sends through the wire.
 // DEPRECATED
 func (a *whisperAdapter) SendPublicRaw(ctx context.Context, chatName string, data []byte) ([]byte, whisper.NewMessage, error) {
-	newMessage := whisper.NewMessage{
+
+	var newMessage whisper.NewMessage
+
+	wrappedMessage, err := a.wrapMessageV1(data)
```

Comment:
Should we respect `sendV1Messages`?

Reply:
`wrapMessageV1` will check the feature flag, it will return the message unchanged or the wrapped payload, the flag will be removed by v1, so I feel checking inside the method means we'll have to remove less code, what do you reckon?

Reply:
Maybe we can call it `tryWrapMessageV1` to indicate that? I agree that it's better to avoid many `if`s but I think we can have here an indicator that it's not definitively a wrapped message but we want to try to wrap it if some conditions are met.

Reply:
sounds good